src/openzaak/components/autorisaties/admin_views.py

```diff
@@ -117,13 +116,17 @@ def get_initial_for_component(
                 if autorisatie.pk in _related_objs_external
             ]
 
-            if spec:
-                _initial["related_type_selection"] = (
-                    RelatedTypeSelectionMethods.all_current_and_future
-                )
-            elif zaaktype_ids == relevant_ids:
-                _initial["related_type_selection"] = (
-                    RelatedTypeSelectionMethods.all_current
+            catalogus_autorisaties_for_va = [
+                catalogus_autorisatie.catalogus.pk
+                for catalogus_autorisatie in catalogus_autorisaties or []
+                if catalogus_autorisatie.max_vertrouwelijkheidaanduiding == va
+            ]
+            if catalogus_autorisaties_for_va:
+                _initial.update(
+                    {
+                        "related_type_selection": RelatedTypeSelectionMethods.select_catalogus,
+                        "catalogi": catalogus_autorisaties_for_va,
+                    }
                 )
             else:
                 _initial.update(
```

Comment:
Hmm you're right, though we will run into another issue because regular Autorisaties and CatalogusAutorisaties are grouped in the same row. For instance, if I save a form with as the first row an autorisatie and in the second row a catalogusautorisatie with the same component/scopes and VA (but different catalogus to avoid overlap). I will only see one or the other because of the form behavior:

![image](https://github.com/user-attachments/assets/1144cf8a-b188-4cba-8114-7b384c2b015f)

I think I'm going to have to refactor this to make CatalogusAutorisaties show in their own rows perhaps?

Reply:
Modified this and now CatalogusAutorisaties display on different rows than regular Autorisaties